pkg/build/api/types.go

```diff
@@ -936,6 +943,43 @@ type BuildLog struct {
 	unversioned.TypeMeta
 }
 
+// DockerStrategyOptions contains extra strategy options for Docker builds
+type DockerStrategyOptions struct {
+	unversioned.TypeMeta
```

Comment:
Why do these include `TypeMeta`?  Are you using them somewhere as a `RawExtension` or a `runtime.Object` directly?